plugins/modules/ndo_virtual_port_channel_interface.py

```diff
@@ -0,0 +1,443 @@
+            interfaces_node_2 = interfaces_node_1
+        else:
+            interfaces_node_2 = ",".join(interfaces_node_2)
+    interface_policy_group = module.params.get("interface_policy_group")
+    interface_policy_group_uuid = module.params.get("interface_policy_group_uuid")
+    interface_descriptions = module.params.get("interface_descriptions")
+    state = module.params.get("state")
+
+    mso_template = MSOTemplate(mso, "fabric_resource", template)
+    mso_template.validate_template("fabricResource")
+
+    existing_virtual_port_channel_interfaces = mso_template.template.get("fabricResourceTemplate", {}).get("template", {}).get("virtualPortChannels", [])
+    object_description = "Virtual Port Channel Interface"
+
+    if state in ["query", "absent"] and not existing_virtual_port_channel_interfaces:
+        mso.exit_json()
+    elif state == "query" and not (name or uuid):
+        mso.existing = existing_virtual_port_channel_interfaces
+    elif existing_virtual_port_channel_interfaces and (name or uuid):
+        match = mso_template.get_object_by_key_value_pairs(
+            object_description,
+            existing_virtual_port_channel_interfaces,
+            [KVPair("uuid", uuid) if uuid else KVPair("name", name)],
+        )
+        if match:
+            virtual_port_channel_attrs_path = "/fabricResourceTemplate/template/virtualPortChannels/{0}".format(match.index)
+            mso.existing = mso.previous = copy.deepcopy(match.details)
+
+    ops = []
+
+    if state == "present":
+        if uuid and not mso.existing:
+            mso.fail_json(msg="{0} with the UUID: '{1}' not found".format(object_description, uuid))
+
+        if interface_policy_group and not interface_policy_group_uuid:
+            fabric_policy_template = MSOTemplate(mso, "fabric_policy", interface_policy_group.get("template"))
+            fabric_policy_template.validate_template("fabricPolicy")
+            interface_policy_group_uuid = fabric_policy_template.get_interface_policy_group_uuid(interface_policy_group.get("name"))
+
+        if mso.existing:
+            mso_values = {
+                "name": name,
+                "description": description,
+                ("node1Details", "node"): node_1,
+                ("node1Details", "memberInterfaces"): interfaces_node_1,
+                ("node2Details", "node"): node_2,
+                ("node2Details", "memberInterfaces"): interfaces_node_2,
+                "policy": interface_policy_group_uuid,
+                "interfaceDescriptions": format_interface_descriptions(mso, interface_descriptions),
+            }
+            append_update_ops_data(ops, match.details, virtual_port_channel_attrs_path, mso_values)
+            mso.sanitize(match.details, collate=True)
+
+        else:
+            mso_values = dict(
+                name=name,
+                description=description,
+                node1Details=dict(
+                    node=node_1,
```

Comment:
When you do not provide `node_1` and `node_2`. I got the below value, and the task failed. Also, check the update part.

```
{
    "op": "add",
    "path": "/fabricResourceTemplate/template/virtualPortChannels/-",
    "value": {
        "description": "Ansible Virtual Port Channel test",
        "interfaceDescriptions": [],
        "name": "ansible_virtual_port_channel_interface",
        "node1Details": {
            "memberInterfaces": "1/1",
            "node": null
        },
        "node2Details": {
            "memberInterfaces": "1/1",
            "node": null
        }
    }
}
```